JetAnalyzers/bin/jet_response_fitter_x.cc

```diff
@@ -255,6 +271,584 @@ int main(int argc,char**argv)
 ////////////////////////////////////////////////////////////////////////////////
 
 //______________________________________________________________________________
+TH1F *hXaxis( TH1F* hist, double minX, double maxX)
+{
+  double *xbins = new double [hist->GetNbinsX()+1];
+  int nbins{0};
+  for(int i=0; i<hist->GetNbinsX(); i++){
+    double x {hist->GetXaxis()->GetBinCenter(i+1)};
+    if(x>=minX && x<=maxX){
+      xbins[nbins]  = hist->GetXaxis()->GetBinLowEdge(i+1);
+      xbins[nbins+1]= hist->GetXaxis()->GetBinUpEdge(i+1);
+      ++nbins;
+    }
+  }
+
+  TH1F *hnew = new TH1F ("hnew", "", nbins, xbins);
+  for(int i=0; i<nbins+1; i++){
+    hnew->SetBinContent(i+1, hist->GetBinContent(hist->FindBin(hnew->GetBinCenter(i+1))));
+    hnew->SetBinError  (i+1, hist->GetBinError  (hist->FindBin(hnew->GetBinCenter(i+1))));
+  }
+  return hnew;
+}
+
+//______________________________________________________________________________
+void fit_modified_fnc_dscb(TH1F*& hrspOrig, string alg, bool calibrated, string histName){
+
+  TH1F* hrsp=(TH1F*)hrspOrig->Clone(); 
+
+
+  // do no write in the output file all the histogram
+  TH1::AddDirectory(kFALSE);
+
+  // select a particular algorithm
+  //if (alg!="ak5tauHPSlooseCombDBcorrAll") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrOneProng0Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrOneProng1Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrTwoProng0Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrTwoProng1Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrThreeProng0Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrThreeProng1Pi0") return;
```

Comment:
Why are all of these commented out? Is it work in progress or does the user need to edit the file each time they want to change algorithms? If the latter, please add this as an option so that we don't need to edit the file and recompile each time.